mongo/integration/initial_dns_seedlist_discovery_test.go

```diff
@@ -0,0 +1,179 @@
+// Copyright (C) MongoDB, Inc. 2017-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package integration
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"path"
+	"runtime"
+	"strings"
+	"testing"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/internal/testutil/assert"
+	"go.mongodb.org/mongo-driver/mongo/integration/mtest"
+	"go.mongodb.org/mongo-driver/x/mongo/driver/connstring"
+	"go.mongodb.org/mongo-driver/x/mongo/driver/description"
+	"go.mongodb.org/mongo-driver/x/mongo/driver/topology"
+)
+
+const (
+	seedlistDiscoveryTestsDir = "../../data/initial-dns-seedlist-discovery"
+)
+
+type seedlistTest struct {
+	URI     string   `bson:"uri"`
+	Seeds   []string `bson:"seeds"`
+	Hosts   []string `bson:"hosts"`
+	Error   bool     `bson:"error"`
+	Options bson.Raw `bson:"options"`
+}
+
+func TestInitialDNSSeedlistDiscoverySpec(t *testing.T) {
+	mtOpts := mtest.NewOptions().Topologies(mtest.ReplicaSet).CreateClient(false)
```

Comment:
The previous runner didn't run the tests unless the `AUTH` env var was set to `noauth`. I don't think this is necessary as the `setSSLSettings` function will skip the test if the environment ssl settings don't match the test requirements.